themes/default/content/blog/mlops-python-iac-deploy-a-chatbot-in-30-minutes/index.md

```diff
@@ -0,0 +1,197 @@
+---
+date: 2023-08-30
+title: "The Real AI Challenge is Cloud, not Code!"
+allow_long_title: true
+meta_desc: "Accelerating AI/ML Innovation with Infrastructure as Code through a Hands-On Journey to deploy your own chatbot."
+meta_image: "chatbot-api-prompt.png"
+authors:
+    - kat-morgan
+tags:
+    - ai
+    - ml
+    - iac
+    - llm
+    - vllm
+    - mlops
+    - azure
+    - llama
+    - llama2
+    - python
+---
+
+The AI industry is stealing the show as tech's goldrush of the '20s. Just looking at ChatGPT's [record setting user growth](https://www.reuters.com/technology/chatgpt-sets-record-fastest-growing-user-base-analyst-note-2023-02-01/), and rapid 3rd party integration [by top brands](https://www.forbes.com/sites/bernardmarr/2023/05/30/10-amazing-real-world-examples-of-how-companies-are-using-chatgpt-in-2023/?sh=ed1c90f14418), it is not surprising the hype suggests this is the beginning of a major digital transformation.
+
+When it comes to taking advantage of AI/ML in your own organization, realizing the benefits comes with major obstacles, not least of which is the almost complete lack of well practiced platform engineering architectures, industry consensus, and operational knowledge. Which is why it is easy to say, *The most difficult challenge that we face with AI today, is a problem of cloud orchestration*.
+
+Now here to help cut through the barriers, in this first installment of the "*Pulumi Python + MLOps*" series we will introduce you to the Katwalk Server, a demo AI application which we will use as a testing ground for developing many parts of the AI development and operations lifecycle.
+
+> # *The most difficult challenge that we face with AI today, is a problem of cloud orchestration*
+
+> figure 1. A sneak peak at the map of ML development and product path
+![A sneak peak at the map of ML development and product path](image.png)
+
+For now, let's save the long winded sales pitches and AI ethics debates for another time. In this post, we will dive into the deep end as we deploy a private LlaMa 2 API chatbot service ... in 30 minutes or less!
+
+> # *deploy a private LlaMa 2 API chatbot service ... in 30 minutes or less!*
+
+Have I mentioned the solution is Python? While Pulumi supports many languages, we chose python for this project as a familiar language in the AI industry.
+
+Let's introduce some key concepts and technologies to start off.
+
+- Infrastructure as Code (IaC) is a transformative approach to platform engineering and orchestration. By applying the rigor and precision of software development practices to cloud operations, IaC offers AI/ML professionals an efficient, reliable, and predictable way to develop at the highest velocity.
```

Comment:
this is the section where you are introducing the solution to the problem. "hey provisioning the infrastructure to run your AIML apps is hard, well IaC is actually a basic way to handle this." run me through the basics of IaC:  version control, testing, etc, etc